File: AlgorithmsPackPythonSolutions/Valid Triangle Number.py
class Solution:
    def triangleNumber(self, nums):
        count = 0
        nums.sort()
        sz = len(nums)
        for i in range(sz - 2):
            k = i + 2
            for j in range(i + 1, sz - 1):
                if nums[i] == 0:
                    break
                while k < sz and nums[i] + nums[j] > nums[k]:
                    k += 1
                count += k - j - 1
        return count

File: AlgorithmsPackJavaSolutions/Valid Triangle Number.java
class Solution {
    public int triangleNumber(int[] nums) {
        int count = 0;
        Arrays.sort(nums);
        int sz = nums.length;
        for (int i = 0; i < sz - 2; i++) {
            int k = i + 2;
            for (int j = i + 1; j < sz - 1; j++) {
                if (nums[i] == 0) {
                    break;
                }
                while (k < sz && nums[i] + nums[j] > nums[k]) {
                    k++;
                }
                count += k - j - 1;
            }
        }
        return count;
    }
}
